src/org/mozilla/javascript/regexp/NativeRegExp.java

```diff
@@ -126,6 +132,36 @@ public static void init(Context cx, Scriptable scope, boolean sealed) {
 
         ctor.setImmunePrototypeProperty(proto);
 
+        ScriptableObject desc = (ScriptableObject) cx.newObject(scope);
```

Comment:
(I don't know much about this area, but...) Could you write simply using Lambda?
https://github.com/mozilla/rhino/blob/ae78dbfbbc50e205e71b19f1d44613ed462879b5/src/org/mozilla/javascript/NativePromise.java#L41-L76

Reply:
Will check.....